modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -545,15 +585,18 @@ struct Infer: public cv::detail::KernelTag {
         // meta order.
         GAPI_Assert(uu.params.input_names.size() == in_metas.size()
                     && "Known input layers count doesn't match input meta count");
-
         for (auto &&it : ade::util::zip(ade::util::toRange(uu.params.input_names),
                                         ade::util::toRange(in_metas))) {
             auto       &&ii = uu.inputs.at(std::get<0>(it));
             const auto & mm =              std::get<1>(it);
 
-            configureInputInfo(ii, mm);
+            configureInputInfo(ii, const_cast<IEUnit*>(&uu), mm);
             ii->getPreProcess().setResizeAlgorithm(IE::RESIZE_BILINEAR);
         }
+        if (!uu.params.reshape_in_names.empty()) {
+            GAPI_Assert(!uu.params.reshape_in_shapes.empty());
+            const_cast<InferenceEngine::CNNNetwork *>(&uu.net)->reshape(uu.params.reshape_in_shapes);
```

Comment:
@TolyaTalamanov, main question about this const_cast for net reshape, other const_cast i can remove.

Reply:
Discussed locally